src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -340,6 +343,57 @@ private void processData(InputData data) {
 		}
 	}
 
+	/**
+	 * NOTE: if a BSSID does not have a non-obsolete entry, it will be returned
+	 * (i.e., it will not be a key in the returned map).
+	 */
+	public Map<String, WifiScanEntry> getAggregatedWifiScans(long obsoletion_period) {
+		Map<String, WifiScanEntry> aggregatedWifiScans = new HashMap<>();
+		for (String bssid : dataModel.latestWifiScans.keySet()) {
+			// flatten the wifiscan entries and sort in reverse chronological order
+			List<List<WifiScanEntry>> scans = dataModel.latestWifiScans.get(bssid);
```

Comment:
Iterate on `map.entrySet()` so that you can call `entry.value()` instead of doing an unneeded lookup